types/shares_test.go

```diff
@@ -175,3 +176,32 @@ func Test_zeroPadIfNecessary(t *testing.T) {
 		})
 	}
 }
+
+func Test_appendToSharesOverwrite(t *testing.T) {
+	var shares NamespacedShares
+
+	// generate some arbitrary first share that must be split
+	newShare := generateRandomNamespacedShares(1, MsgShareSize+1)[0]
+
+	// make a copy of the portion of the share to check if it's overwritten later
+	extraCopy := make([]byte, MsgShareSize)
+	copy(extraCopy, newShare.Share[:MsgShareSize])
+
+	// use appendToShares to add our new share
+	appendToShares(shares, newShare.ID, newShare.Share)
+
+	// check if the original share data has been overwritten.
+	assert.Equal(t, extraCopy, []byte(newShare.Share[:MsgShareSize]))
+}
```

Comment:
this test fails without the added copy in `split`

Reply:
Are you sure it is the copy or simply the fact that previously the nid got prefixed? `rawShare := []byte(append(nid, rawData[:shareSizeOrLen]...))`? What happens if this is changed to `rawShare := rawData[:shareSizeOrLen]` instead?

Reply:
ahh I see, it works!